scripts/pylib/twister/twisterlib/testsuite.py

```diff
@@ -244,6 +263,18 @@ def _find_ztest_testcases(search_area, testcase_regex):
     """
     testcase_regex_matches = \
         [m for m in testcase_regex.finditer(search_area)]
+
+    a = {}
+    for m in testcase_regex_matches:
+        try:
+            s = m.group("suite_name")
+        except IndexError:
+            continue
+        tc = m.group("testcase_name").decode("UTF-8").replace("test_", "", 1)
+        tests = a.get(s, [])
```

Comment:
Probably should be:
```python
tests = a.get(s.decode("UTF-8"), [])
```
or even better, add `decode("UTF-8")` few lines earlier when you create `s` variable:
```python
s = m.group("suite_name").decode("UTF-8")
```